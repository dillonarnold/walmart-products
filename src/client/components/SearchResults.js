import React from 'react';
import { shape, string, number, arrayOf } from 'react-proptypes';
import { connect } from 'react-redux';
import Product from './Product';

const propTypes = {
  products: arrayOf(shape({
    itemId: number,
    name: string,
    mediumImage: string
  }))
};

const defaultProps = {
  products: []
};

const SearchResults = ({products}) => (
    <div id="product-list">
      {
        products.map((product) =>
          <Product product={product} key={product.itemId}/>
        )
      }
    </div>
);

const mapStateToProps = state => {
  return {
    products: state.products
  };
};


SearchResults.propTypes = propTypes;
SearchResults.defaultProps = defaultProps;

export default connect(mapStateToProps)(SearchResults);