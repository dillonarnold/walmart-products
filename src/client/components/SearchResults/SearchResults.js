import React from 'react';
import { shape, string, number, arrayOf, bool } from 'react-proptypes';
import CircularProgress from '@material-ui/core/CircularProgress';
import Grid from '@material-ui/core/Grid';
import Product from '../Product/Product';

const propTypes = {
  /** Array of products */
  products: arrayOf(shape({
    /** Id of the product */
    itemId: number,
    /** Name of the product */
    name: string,
    /** The medium image src of the product */
    mediumImage: string
  })),
  /** Indicates if we are fetching products */
  loading: bool
};

const defaultProps = {
  products: []
};

/**
 * Takes an array of products and displays a list of Product components.
 *
 * @version 1.0.0
 * @author Dillon Arnold
 */
const SearchResults = ({products, loading}) => (
  <div id="search-results">
  {
    loading ? (
      <Grid container justify="center" spacing={24}>
        <Grid item md={12}>
          <CircularProgress size={100} />
        </Grid>
      </Grid>
    ) : (
      <div id="product-list">
        {
          products.map((product) =>
            <Product product={product} key={product.itemId}/>
          )
        }
      </div>
    )
  }
  </div>
);

SearchResults.propTypes = propTypes;
SearchResults.defaultProps = defaultProps;

export default SearchResults;