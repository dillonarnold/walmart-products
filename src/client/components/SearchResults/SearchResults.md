Example:

```jsx

const products = [
	{
		itemId: 123456, 
		name: "iPod", 
		mediumImage: "https://i5.walmartimages.com/asr/cecd7151-08ae-47ce-84f2-b46abda0f5ca_1.22789781b5d7d5632eea50d3789f2156.jpeg?odnHeight=180&odnWidth=180&odnBg=FFFFFF"
	},
	{
		itemId: 321314, 
		name: "iPod Case", 
		mediumImage: "https://i5.walmartimages.com/asr/663a6e96-81e3-40be-b35c-8008fd3fcc5f_1.c250437ad5130eaf00254fc27285474d.jpeg?odnHeight=180&odnWidth=180&odnBg=FFFFFF"
	}
];

<SearchResults products={products} />
```