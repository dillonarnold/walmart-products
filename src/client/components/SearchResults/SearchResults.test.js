import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import SearchResults from './SearchResults';

describe('Product component', () => {
  const results = [
    {
      itemId: 123456,
      name: 'iPod',
      mediumImage: 'imageSrc'
    },
    {
      itemId: 231313,
      name: 'iPod Case',
      mediumImage: 'imageSrc2'
    }
  ];

  it('should render properly', () => {
    const component = shallow(
      <SearchResults producst={results} />,
    );

    expect(toJson(component)).toMatchSnapshot();

  });

  it('test props', () => {
    const component = mount(<SearchResults products={results} />);
    expect(component.prop('products')).toEqual(results);
  });
});
