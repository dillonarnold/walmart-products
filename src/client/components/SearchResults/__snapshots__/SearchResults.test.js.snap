// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product component should render properly 1`] = `
<div
  id="search-results"
>
  <div
    id="product-list"
  />
</div>
`;
